test/test_sparse.py

```diff
@@ -3248,33 +3025,146 @@ def test_cuda_sparse_cpu_dense_add(self):
         with self.assertRaisesRegex(RuntimeError, "add: expected 'self' to be a CUDA tensor, but got a CPU tensor"):
             x + sparse_y
 
+
 class TestSparseUnaryUfuncs(TestCase):
     exact_dtype = True
 
-    @ops(sparse_unary_ufuncs)
-    def test_sparse_consistency(self, device, dtype, op):
-        unsupportedTypes = [torch.bfloat16, torch.cfloat, torch.cdouble]
+    def _skip_helper(self, dtype):
+        unsupportedTypes = [torch.bfloat16, torch.float16, torch.bool, torch.complex64, torch.complex128]
         if dtype in unsupportedTypes:
-            self.skipTest('Skipped! Unsupported dtypes for Sparse')
+            self.skipTest(f'Skipped! "coalesce" not implemented for {dtype}')
+
+    def _get_dense_and_sparse_samples(self, device, dtype, op):
+        test_backward = op.test_complex_grad or not dtype.is_complex
```

Comment:
We also have supports_autograd now:

https://github.com/pytorch/pytorch/blob/e3a805b9c5fceeb1bd58d7cd309271f513386595/torch/testing/_internal/common_methods_invocations.py#L164